netlify/functions/manage-bot.ts

```diff
@@ -0,0 +1,363 @@
+    paymentAddresses,
+  };
+
+  let avatarFilename = "";
+  let base64Avatar = "";
+  if (image) {
+    const decodedImage = Buffer.from(image, "base64");
+    const { data, info } = await sharp(decodedImage)
+      .resize({ width: 512 })
+      .toBuffer({ resolveWithObject: true });
+    base64Avatar = data.toString("base64");
+    avatarFilename = `${botName}.${info.format}`;
+    newBotData.image = `./avatars/${botName}.${info.format}`;
+  }
+
+  if (relegated?.newValue !== undefined) {
+    newBotData.relegated = relegated.newValue;
+  }
+
+  // Don't create a PR if the json content and image have not changed
+  if (isEqual(newBotData, oldBotData) && !image) {
+    return {
+      statusCode: 422,
+      body: JSON.stringify({ error: "Nothing changed" }),
+    };
+  }
+
+  const files: { [path: string]: string | File } = {
+    [`_data/bots/${botName}.json`]: JSON.stringify(newBotData, null, 2),
+  };
+  if (image) {
+    files[`_data/bots/avatars/${avatarFilename}`] = {
+      content: base64Avatar,
+      encoding: "base64",
+    };
+  }
+
+  const title = `Edit bot ${botName}`;
+  const body = dedent`
+  This auto-generated PR modifies the bot ${botName}
+
+  The request was made by ${username}
+  `;
+  const branchName = `bot/${botName}`;
+
+  try {
+    const res = await octokit.createPullRequest({
+      owner: process.env.GITHUB_REPO_OWNER!,
+      repo: process.env.REPO!,
+      title,
+      body,
+      head: branchName,
+      base: process.env.BRANCH_NAME,
+      changes: [
+        {
+          files,
+          commit: title,
+        },
+      ],
+    });
+
+    if (!res) {
+      return {
+        statusCode: 400,
+        body: JSON.stringify({ error: "Failed to create pull request." }),
+      };
+    }
+
+    // submit helpdesk ticket
+    await fetch(`${process.env.URL}/.netlify/functions/request-support`, {
+      method: "POST",
+      headers: {
+        "Content-Type": "application/json",
+        "X-Authorization": `Bearer ${sessionToken}`,
+        "C4-User": username,
+      },
+      body: JSON.stringify({
+        subject: "Bot Update",
+        request: "team",
+        description: `PR link: ${res.data.html_url}`,
+        discordHandle: username,
+      }),
+    });
+
+    const newMemberEmails = await getGroupEmails(newBotData.crew);
+    const oldMemberEmails = await getGroupEmails(oldBotData.crew);
+    const emailSubject = `Code4rena team "${botName}" has been modified`;
+
+    const emailBody = dedent`
+      Changes to the team ${botName} have been requested.
+
+      Members: ${crew.newValue.join(", ")}
+
+      Polygon address: ${polygonAddress.newValue}
+      ${
+        ethereumAddress &&
+        (ethereumAddress.newValue || ethereumAddress.oldValue)
+          ? "Ethereum address: " + ethereumAddress.newValue
+          : ""
+      }
+
+      You can see the PR here: ${res.data.html_url}
+    `;
+    await sendConfirmationEmail(
+      [...newMemberEmails, ...oldMemberEmails],
```

Comment:
does the mail service de-dupe these? (now I'm curious what happens if I send multiple myselfs the same email in general :-)

Reply:
Yes it does - using lodash/uniq